src/others/observable/lv_observable.h

```diff
@@ -0,0 +1,263 @@
+ * @param buf string to copy 
+ * @note automatically notifies all observers
+ */
+void lv_subject_copy_string(lv_subject_t* subject, char* buf);
+
+/**
+ * Set bitmask and value comparison pattern of the observer
+ * @param observer address of the observer
+ * @param pattern pattern used to mask out interested bits from the bitmask's value and compare it with a constant value
+ * @note The value
+ */
+void lv_observer_bitmask_set_cond(lv_observer_t* observer, lv_bitmask_pattern_t pattern);
+
+/**
+ * Bind bitmask to object flag
+ * @param subject address of the subject
+ * @param obj the LVGL object what will be affected by the change
+ * @param flag the affected flag
+ * @param pattern pattern used to mask out interested bits from the bitmask's value and compare it with a constant value
+ */
+void* lv_bind_bitmask_to_obj_flag(lv_subject_t* subject, lv_obj_t* obj, lv_obj_flag_t flag, lv_bitmask_pattern_t pattern);
+
+/**
+ * Bind bitmask to object state
+ * @param subject address of the subject
+ * @param obj the LVGL object what will be affected by the change
+ * @param state the affected state
+ * @param pattern pattern used to mask out interested bits from the bitmask's value and compare it with a constant value
+ */
+void* lv_bind_bitmask_to_obj_state(lv_subject_t* subject, lv_obj_t* obj, lv_state_t flag, lv_bitmask_pattern_t pattern);
+
+/**
+ * Bind int value to a local style property
+ * @param subject address of the subject
+ * @param obj the LVGL object what will be affected by the change
+ * @param prop the style property
+ * @param selector the part/state selector
+ */
+void* lv_bind_int_to_obj_local_style_prop(lv_subject_t* subject, lv_obj_t* obj, lv_style_prop_t prop, lv_style_selector_t selector);
+
+/**
+ * Bind string value to a label text
+ * @param subject address of the subject
+ * @param obj the LVGL object what will be affected by the change
+ */
+void* lv_bind_string_to_label_text(lv_subject_t* subject, lv_obj_t* obj);
+
+/**
+ * Bind int value to a label text using a predefined format string
+ * @param subject address of the subject
+ * @param obj the LVGL object what will be affected by the change
+ * @param fmt the format string
+ */
+void* lv_bind_int_to_label_text_fmt(lv_subject_t* subject, lv_obj_t* obj, const char* fmt);
+
+/**
+ * Bind and int value to a label text using a variable format string
+ * @param subject address of the subject
+ * @param fmt the address of the format string (an observable string value)
+ * @param obj the LVGL object what will be affected by the change
+ */
+void* lv_bind_formatted_int_to_label_text(lv_subject_t* subject, lv_subject_t* fmt, lv_obj_t* obj);
+
+/**
+ * Generic method to subscribe to an int value with a user-defined callback
+ * @param subject address of the subject
+ * @param cb the callback called when the subject changes
+ * @param obj the LVGL object what will be affected by the change
+ * @param data1 optional additional parameter stored in the observer object
```

Comment:
For the MicroPython binding we will need a `user_data` field in `lv_subject_t`. I wonder if we can use only a single `user_data` instead of 2 `data`.

Reply:
Could you be more specific? What type should user_data be? How can you replace the two variables with one without dynamic allocation?

Reply:
It should be `void * user_data`.

We can do dynamic allocation :)

Reply:
Yes, we can, but shall we? In most cases there is only a single parameter. We only have two parameters (so far) for style properties. Do we want to allocate memory for a single int as well? It seems like an overkill to me.
We could also implement some kind of inheritance, so that we could derive new "classes" with additional data if needed. I don't know how this would work with MicroPython though.